docs/docs/gatsby-lifecycle-apis.md

```diff
@@ -13,4 +13,49 @@ Gatsby's design principles include:
   trivial and encouraged.
 - Plugins are easy to open source and reuse. They're just NPM packages.
 
-See the links along the left under "REFERENCE" for the full API documentation.
+# High level Overview
+
+## Bootstrap sequence
+
+During "bootstrap" gatsby:
+
+- reads `gatsby-config.js` to load in your list of plugins
+- initializes its cache (stored in `/.cache`)
+- pulls in and preprocesses data ("source and transform nodes") into a GraphQL schema
+- creates pages in memory
+  - from your `/pages` folder 
+  - from your `gatsby-node.js` if you implement `createPages`/`createPagesStatefully` (e.g. templates)
+  - from any plugins that implement `createPages`/`createPagesStatefully`
+- extracts, runs, and replaces graphql queries for pages and `StaticQuery`s
+- writes out the pages to cache
+
+In development this is a running process powered by [Webpack](https://github.com/gatsbyjs/gatsby/blob/dd91b8dceb3b8a20820b15acae36529799217ae4/packages/gatsby/package.json#L128) and [react-hot-loader](https://github.com/gatsbyjs/gatsby/blob/dd91b8dceb3b8a20820b15acae36529799217ae4/packages/gatsby/package.json#L104), so changes to any files get re-run through the sequence again, with smart cache invalidation.
+
+The core of the bootstrap process is the "api-runner", which helps to execute APIs in sequence, with state managed in Redux. Gatsby exposes a number of lifecycle APIs which can either be implemented by you or any of your provided plugins in `gatsby-node.js`, `gatsby-browser.js` or `gatsby-ssr.js`. **The sequence of your plugins matter**, so if the output of one plugin depends on another you should be sure to reflect this in `gatsby-config.js`.
```

Comment:
The sequence of plugins should _not_ matter. If they do, it's a bug. Plugins shouldn't ever depend on each other. If they _do_, there should be an explicit hierarchal relationship between the plugins e.g. gatsby-transformer-remark and its family of gatsby-remark-* plugins.

Reply:
I mean.. respectfully... these two conflict:

Reply:
i realize this can touch on some unresolved questions in gatsby itself - im happy to simply omit the sentence. but i just wrote that based on what i saw in the issues, just trying to help not misinform.

Reply:
By design it's very hard but _not_ impossible for plugins to depend on each other. So occasionally there can be implicit dependencies but these should be treated as bugs and fixed.

Reply:
hmm: example to maybe motivate discussion https://github.com/gatsbyjs/gatsby/issues/5196#issuecomment-386318952